falcon/asgi/request.py

```diff
@@ -345,13 +345,28 @@ class Request(falcon.request.Request):
 
     __slots__ = [
         '_asgi_headers',
-        '_asgi_server_cached',
+        # '_asgi_server_cached',
         '_first_event',
         '_receive',
-        '_stream',
+        # '_stream',
         'scope',
     ]
 
+    # PERF(vytas): Dirty hack to fall back to class attributes when unset;
```

Comment:
Not sure I would classify this as an hack

Reply:
I would tune down the language regardless, if we go forward with the approach.

Reply:
Reworded to a more marketable version.